module_utils/ns1.py

```diff
@@ -65,3 +70,25 @@ def errback(args):
             self.module.fail_json(msg="%s - %s" % (args[0], args[1]))
 
         return errback
+
+
+class Decorators(object):
+    @classmethod
+    def skip_in_check_mode(self, func):
+        """Decorater function that skips passed function if module is
+        in check_mode.  If module is not in check_mode, passed function
+        executes normally.
+
+        :param func: Function to wrap
+        :type func: func
+        :return: Wrapped function
+        :rtype: func
+        """
+
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            if self.module.check_mode:
+                return
+            return func(self, *args, **kwargs)
+
+        return wrapper
```

Comment:
this is new but I don't see it being used? Also curious why you wrapped it in a class

Reply:
This is used as a decorator on methods that make API calls in `ns1_zone.py` (i.e. https://github.com/ns1/ns1-ansible-modules/blob/ad0eb8a4e74610d847b5f933f7571e036bc3f432/library/ns1_zone.py#L324).  Can be reused when we refactor `ns1_record` and in new modules.

Shoved it in a class as a single place to keep our decorators in case we add more down the line (i.e. to make imports easier, etc).  Originally had it as class method of `NS1ModuleBase` but that opened a whole can of worms of issues I didn't want to deal with.